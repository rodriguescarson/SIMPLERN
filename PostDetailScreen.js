// PostDetailScreen.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const PostDetailScreen = ({ route }) => {
  const { post } = route.params;

  return (
      <View testID="post-detail" style={styles.container}>
          <Text style={styles.id}>Post Id:{post.id}</Text>
          <Text style={styles.id}>User Id:{post.userId}</Text>
          <Text style={styles.title}>{post.title}</Text>
          <Text style={styles.body}>{post.body}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
    marginBottom: 20,
  },
  body: {
    fontSize: 18,
    },
    id: {
        fontSize: 12,
        fontWeight: 'bold',
        marginBottom: 20,
  }
});

export default PostDetailScreen;
