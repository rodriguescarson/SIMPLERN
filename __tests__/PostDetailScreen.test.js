import React from 'react';
import { render } from '@testing-library/react-native';
import PostDetailScreen from '../PostDetailScreen';

describe('PostDetailScreen', () => {
  const post = {
    id: 1,
    userId: 1,
    title: 'Example Title',
    body: 'Example Body',
  };

  it('renders the PostDetailScreen component', () => {
    const { getByText, getByTestId } = render(<PostDetailScreen route={{ params: { post } }} />);

    // Assert the presence of specific elements and their content
    expect(getByTestId('post-detail')).toBeTruthy();
    expect(getByText(`Post Id:${post.id}`)).toBeTruthy();
    expect(getByText(`User Id:${post.userId}`)).toBeTruthy();
    expect(getByText(post.title)).toBeTruthy();
    expect(getByText(post.body)).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const { toJSON } = render(<PostDetailScreen route={{ params: { post } }} />);

    // Use snapshots to compare the rendered component with a saved snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
