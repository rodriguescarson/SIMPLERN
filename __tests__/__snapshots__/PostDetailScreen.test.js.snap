// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetailScreen matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
  testID="post-detail"
>
  <Text
    style={
      {
        "fontSize": 12,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Post Id:
    1
  </Text>
  <Text
    style={
      {
        "fontSize": 12,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    User Id:
    1
  </Text>
  <Text
    style={
      {
        "fontSize": 28,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Example Title
  </Text>
  <Text
    style={
      {
        "fontSize": 18,
      }
    }
  >
    Example Body
  </Text>
</View>
`;
